repose-aggregator/core/repose-core/src/main/java/org/openrepose/core/services/rms/ResponseMessageServiceImpl.java

```diff
@@ -93,10 +96,19 @@ public boolean isInitialized() {
     }
 
     @Override
-    public void handle(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    public void handle(HttpServletRequest request, HttpServletResponseWrapper response) throws IOException {
         final StatusCodeMatcher matchedCode = getMatchingStatusCode(String.valueOf(response.getStatus()));
-        final MutableHttpServletRequest mutableRequest = MutableHttpServletRequest.wrap(request);
-        MediaRangeProcessor processor = new MediaRangeProcessor(mutableRequest.getPreferredHeaders("Accept", DEFAULT_TYPE));
+        final HttpServletRequestWrapper wrappedRequest = new HttpServletRequestWrapper(request);
```

Comment:
I was trying to scope where we wrap the request and response. I want it to be clear where we actually need the wrapper, like in this case, where we're only using it to gain access to the `getPreferredHeaders(...)` method. If we call this method conditionally, why should we wrap the request outside of this method?

On the whole, scoping things in this way give us less overhead, and I like the encapsulation.